imap_processing/codice/codice_l1a.py

```diff
@@ -193,26 +192,37 @@ def define_data_variables(self) -> xr.Dataset:
         # Stack the data so that it is easier to reshape and iterate over
         all_data = np.stack(self.data)
 
-        # The dimension of all data is (epoch, num_counters, num_energy_steps,
-        # num_positions, num_spin_sectors) (or may be slightly different
-        # depending on the data product). In any case, iterate over the
-        # num_counters dimension to isolate the data for each counter so
-        # that it can be placed in a CDF data variable.
+        # The dimension of all_data is something like (epoch, num_counters,
+        # num_energy_steps, num_positions, num_spin_sectors) (or may be slightly
+        # different depending on the data product). In any case, iterate over
+        # the num_counters dimension to isolate the data for each counter so
+        # each counter's data can be placed in a separate CDF data variable.
         for counter, variable_name in zip(
             range(all_data.shape[1]), self.config["variable_names"]
         ):
-            counter_data = all_data[:, counter, :, :, :]
+            # Dynamically determine the slicing required to extract the counter
+            # data, since the number of dimensions may vary. The counter
+            # dimension is always the first
+            slicing: list[Any] = [slice(None)] * all_data.ndim
+            slicing[1] = counter
+
+            # Extract the counter data
+            counter_data = all_data[tuple(slicing)]
```

Comment:
This is pretty janky and I couldn't think of a better way to do this. Happy to hear if anyone has better ideas.

Reply:
I'm not sure what exactly you're trying to do here, but perhaps using an elipses?
`counter_data = all_data[:, counter, ...]`

Reply:
Thanks, this is much better 😌